src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java

```diff
@@ -1,44 +1,279 @@
+            int eventsCount = 0;
+
+            for (String eventName : eventNames) {
+                eventsCount++;
+                JsonNode eventJson = parsedJSON.get(eventName);
+                String event = eventJson.toString();
+                rmqHandler.publishObjectToWaitlistQueue(event);
+            }
+
+            assert(waitForEventsToBeProcessed(eventsCount));
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+        LOGGER.debug("Eiffel events sent.");
+    }
+
+    private void readSubscriptionNames(String readFileToString) {
+        try {
+            JSONArray array = new JSONArray(readFileToString);
+            for (int i = 0; i < array.length(); i++) {
+                subscriptionNames.add(array.getJSONObject(i).get("subscriptionName").toString());
+            }
+        } catch (JSONException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
     }
 
-    @Given("^Subscription mail trigger is created$")
-    public void subscription_mail_trigger_is_created() throws Throwable {
-        LOGGER.debug("Creating subscription mail trigger");
+    private void postSubscriptions(String readFileToString, String endPoint) throws Exception {
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.post(endPoint).accept(MediaType.APPLICATION_JSON)
+                .content(readFileToString).contentType(MediaType.APPLICATION_JSON);
+
+        postResult = mockMvc.perform(requestBuilder).andReturn();
+        LOGGER.debug("Response code from REST when adding subscriptions: "
+                + String.valueOf(postResult.getResponse().getStatus()));
+
+        assertEquals(HttpStatus.OK.value(), postResult.getResponse().getStatus());
+    } 
+
+    private void validateSubscriptionsSuccessfullyAdded(String endPoint) throws Exception {
+        RequestBuilder getRequest = MockMvcRequestBuilders.get(endPoint);
+        getResult = mockMvc.perform(getRequest).andReturn();
+
+        LOGGER.debug("Response code from REST when getting subscriptions: "
+                + String.valueOf(getResult.getResponse().getStatus()));
+        assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
+
+        LOGGER.debug("Checking that response contains all subscriptions");
+        for (String subscriptionName : subscriptionNames) {
+            assertTrue(getResult.getResponse().getContentAsString().contains(subscriptionName));
+        }        
     }
 
-    @Given("^Subscription rest authenticated trigger is created$")
-    public void subscription_rest_authenticated_trigger_is_created() throws Throwable {
-        LOGGER.debug("Subscription rest authenticated trigger is created");
+    private boolean requestBodyContainsStatedValues(JSONArray jsonArray) throws JSONException {
+        int tc5 = 0, successfull = 0;
+        for(int i = 0; i < jsonArray.length(); i++){
+            if (jsonArray.getString(i).contains("TC5")) {
```

Comment:
save jsonArray.getString(i) in a local variable instead of fetching it 2 times

Reply:
Done.